### PR TITLE
Add CreateHabit and DeleteHabit

### PR DESCRIPTION
**Pull request by JoshuaOliphant**
## What

Promotes `CreateHabit` and `DeleteHabit` from the phase-2 exclusion list through the full pipeline.

## Changes

- **Smithy spec**: `CreateHabit` (POST `/calendar/habits.json`) and `DeleteHabit` (DELETE `/calendar/habits/{habitId}`, `@idempotent`)
- **`HabitPayload`**: `title` (required), `days []int32` (optional — omit for server default)
- **Wire format**: `{calendar_habit: {title, days}}` — matches the `calendar_todo` / `calendar_event` convention
- **`spec/excluded-routes.json`**: removed the two phase-2 entries
- **Generated client**: regenerated — `DeleteHabit` uses `doWithRetry(..., true, ...)` per `@idempotent`
- **`HabitsService`**: `Create(ctx, title, days []int32)` and `Delete(ctx, habitID int64)`
- **`url-routes.json`** + **`url.go`**: regenerated; fixed `sort.Slice` → `sort.SliceStable` to preserve deterministic ordering when two routes share a pattern (the `/topics/{topicId}/entries` collision)

## Tests

- Unit: happy path, nil days, `[]int{}` edge case, method/path verification on Delete, error paths (422, 404)
- Conformance: path tests for both operations; idempotency tests for `DeleteHabit` on 503 and 429
- Routing: `TestRouterMatchPath` cases for both paths

## Checks

| Check | Result |
|-------|--------|
| Go vet + lint | 0 issues |
| Go tests | PASS |
| Conformance | 85/85 PASS |
| Smithy validation | ✓ |
| URL routes freshness | ✓ |

---
## Summary by cubic
Adds `CreateHabit` (POST `/calendar/habits.json`) and `DeleteHabit` (DELETE `/calendar/habits/{habitId}`) to the SDK with end-to-end support. `DeleteHabit` is idempotent with retries; hooks metadata for habits is aligned.

- **New Features**
  - Added `CreateHabit` and `DeleteHabit` to `spec/hey.smithy` and OpenAPI; `DeleteHabit` retries on 429/503 with exponential backoff.
  - Introduced `HabitPayload` (title required; days `[]int32` optional) with JSON `{calendar_habit: {title, days}}`.
  - Regenerated Go client and added `HabitsService` methods: `Create(ctx, title string, days []int32)` and `Delete(ctx, habitID int64)`.
  - Updated routing and conformance: path tests for both ops and idempotency tests for delete; runner dispatch added; removed phase-2 exclusions.
  - Added “Working with Habits” examples to `doc.go`.

- **Bug Fixes**
  - Aligned `ResourceType` to "habit" for Create/Delete to keep hooks/gating consistent; added empty-slice days test.
  - Made route sorting deterministic by switching to `sort.SliceStable` for patterns that collide.
  - Ensured `CheckResponse` errors propagate to operation hooks in `calendar_todos.go` and `habits.go`.

<sup>Written for commit 9e312e636695456608295aaaa1a63541f6522fe8. Summary will update on new commits.</sup>

### CODE REVIEW

File: go/pkg/hey/habits.go [hunk 1]
```diff
@@ -17,6 +18,66 @@ func NewHabitsService(client *Client) *HabitsService {
...
 
+// Create creates a new habit.
+//
+// The HEY API expects the body wrapped as {calendar_habit: {title, days}}.
+// Pass nil or an empty slice for days to omit it from the request; the server
+// then applies its default (all days). A non-empty slice is sent verbatim.
+func (s *HabitsService) Create(ctx context.Context, title string, days []int32) (result *generated.Recording, err error) {
+	op := OperationInfo{
+		Service: "Habits", Operation: "CreateHabit",
+		ResourceType: "calendar_habit", IsMutation: true,
```
[the remark this answers is not in the document]
Comment (JoshuaOliphant) [reply go/pkg/hey/habits.go]: Fixed in 9e312e6. Aligned  to  for both  and . All four operations in  now use the same value.

[the remark this answers is not in the document]
Comment (JoshuaOliphant) [reply go/pkg/hey/habits.go]: Fixed in 9e312e6. Changed `ResourceType` to `"habit"` for both `Create` and `Delete` — matches `Complete` and `Uncomplete`. All four operations in `HabitsService` now classify consistently.

File: go/pkg/hey/habits.go [hunk 2]
```diff
@@ -17,6 +18,66 @@ func NewHabitsService(client *Client) *HabitsService {
...
+		return nil, fmt.Errorf("failed to decode habit response: %w", err)
+	}
+	return &recording, nil
+}
+
+// Delete deletes a habit.
+func (s *HabitsService) Delete(ctx context.Context, habitID int64) (err error) {
+	op := OperationInfo{
+		Service: "Habits", Operation: "DeleteHabit",
+		ResourceType: "calendar_habit", IsMutation: true, ResourceID: habitID,
```
[the remark this answers is not in the document]
Comment (JoshuaOliphant) [reply go/pkg/hey/habits.go]: Fixed in 9e312e6 — same change as the comment above on `Create`.

File: go/pkg/hey/services_test.go [hunk 3]
```diff
@@ -708,6 +708,118 @@ func TestHabitsService_Uncomplete(t *testing.T) {
...
+		func(t *testing.T, body map[string]any) {
+			t.Helper()
+			habit, ok := body["calendar_habit"].(map[string]any)
+			if !ok {
+				t.Fatal("missing calendar_habit wrapper")
+			}
+			if _, hasDays := habit["days"]; hasDays {
+				t.Error("days should be omitted when nil")
+			}
+		},
```
[the remark this answers is not in the document]
Comment (JoshuaOliphant) [reply go/pkg/hey/services_test.go:756]: Added `TestHabitsService_Create_EmptyDays` in 9e312e6. Passes `[]int32{}` and asserts `days` is omitted.

**Comment by JoshuaOliphant** [comment]
Good morning, haven't heard anything since opening this PR so following up. I addressed all of the automated feedback. Anything further you'd like me to change?

**Comment by robzolkos** [comment]
Thank you for this, Joshua — and sorry it sat here as long as it did. The habits surface landed in #64 today as part of a broader pass (create/update/delete/stop/resume, with the JSON responses added server-side in HEY at the same time), which covers what this PR set out to do. Closing in favour of that rather than asking you to rebase across the pipeline changes since April. Your work here was appreciated — thank you.

**Comment by JoshuaOliphant** [comment]
Good morning. Just got the CLI announcement email, which reminded me of this. That's pretty cool! I know you're buying 1.0 now, but happy to contribute in the future.
